File: 백준/Silver/10828.스택/스택.java
import java.io.BufferedReader;
import java.io.BufferedWriter;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;
import java.io.IOException;
import java.util.StringTokenizer;
import java.util.ArrayList;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in)); 
        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));  
        
        int N = Integer.parseInt(br.readLine());
        
        Stack ans = new Stack();

        for (int i = 0; i < N; i++) {
            StringTokenizer st = new StringTokenizer(br.readLine());
            String mtd = st.nextToken();
    
            int answer = 0;
    
            if (mtd.equals("push")) {
                int M = Integer.parseInt(st.nextToken()); 
                ans.push(M);
                continue;
            }
            else if (mtd.equals("pop")) answer = ans.pop();
            else if (mtd.equals("size")) answer = ans.size();
            else if (mtd.equals("empty")) answer = ans.empty();
            else if (mtd.equals("top")) answer = ans.top();
            
            bw.write(answer + "\n");
            bw.flush();
        }
    }
}

class Stack {
    ArrayList<Integer> stck = new ArrayList<>();
    
    public void push(int x) {
        stck.add(x);
    }

    public int pop() {
        int out;
        if (stck.size() > 0) 
            out = stck.remove(stck.size()-1);
        else
            out = -1;
        return out;
    }

    public int size() {
        return stck.size();
    }

    public int empty() {
        if (stck.size() > 0)
            return 0;
        else return 1;
    }

    public int top() {
        int out;
        if (stck.size() > 0) 
            out = stck.get(stck.size()-1);
        else
            out = -1;
        return out;
    }
}

File: 백준/Silver/10828.스택/스택.cc
#include <cstdio>
#include <algorithm>
#include <stack>
#include <cstring>
using namespace std;



int N;
stack<int> st;
char str[10];
int main(){
    scanf("%d",&N);
    for(int i=1;i<=N;i++){
        scanf("%s",str);
        if(!strcmp(str,"push")){
            int a;
            scanf("%d",&a);
            st.push(a);
        }else if(!strcmp(str,"pop")){
            if(st.size()==0){
                printf("-1\n");
                continue;
            }
            printf("%d\n",st.top());
            st.pop();
        }else if(!strcmp(str,"top")){
            if(st.size()==0){
                printf("-1\n");
                continue;
            }
            printf("%d\n",st.top());
        }else if(!strcmp(str,"empty")){
            printf("%d\n",st.empty()?1:0);
        }else if(!strcmp(str,"size")){
            printf("%d\n",st.size());
        }
    }
    return 0;
}
